services/dns/dns.proto

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2022 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+syntax = "proto3";
+
+option go_package = "github.com/Snowflake-Labs/sansshell/dns";
+
+package Dns;
+
+// The Dns service definition
+service Lookup {
+  // Lookup issues a DNS lookup request and returns the result
+  rpc Lookup(LookupRequest) returns (LookupReply) {}
+}
+
+// LookupRequest describes the DNS query request
+message LookupRequest {
+  // The hostname to query
+  string hostname = 1;
+}
+
+// LookupReply describes the result of a DNS query
+message LookupReply {
+  bytes result = 1;
```

Comment:
why not a string or if this is N of them a repeated string?

Reply:
Initially I thought about the `dig` output format but it's much cleaner to not fiddle around with a byte array here, changed in 3f31c60002e0aba6c8a2ae49ccabc7211d1db494